### PR TITLE
Refactor auth logic for `returnTo` params

### PR DESCRIPTION
**Pull request by ProchaLu**
This PR updates the profile screen and single guest screen to check for the user information. If the user is not authenticated, they are redirected to the login page, with a `returnTo` parameter included to direct them back to their intended screen after successful login.

Additionally, this PR updates the 2 fetch functions to a single `Promise.all` in the guests tab.

### CODE REVIEW

File: app/(tabs)/profile.tsx [hunk 1]
```diff
@@ -41,14 +43,13 @@ export default function Profile() {
         const body: UserResponseBodyGet = await response.json();
 
         if ('error' in body) {
-          Alert.alert('Error', body.error, [{ text: 'OK' }]);
-          return router.push('/(auth)/login');
+          router.replace('/(auth)/login?returnTo=/(tabs)/profile');
```
Comment (karlhorky) [inline app/(tabs)/profile.tsx:46]: let's match the pattern from above, even though we don't need the `return`

```suggestion
          router.replace('/(auth)/login?returnTo=/(tabs)/profile');
          return;
```

File: app/guests/[guestId].tsx [hunk 2]
```diff
@@ -115,11 +116,26 @@ export default function GuestPage() {
...
 
   useFocusEffect(
     useCallback(() => {
+      async function getUser() {
+        const response = await fetch('/api/user');
+
+        const body: UserResponseBodyGet = await response.json();
+
+        if ('error' in body) {
+          router.replace(`/(auth)/login?returnTo=/(tabs)/guests`);
```
[the remark this answers is not in the document]
Comment (ProchaLu) [reply app/guests/[guestId].tsx]: [240a21e](https://github.com/upleveled/expo-example-fall-2024-atvie/pull/3/commits/240a21ecec9d090bce65355c5b81f5b0f6c175db)

**Comment by karlhorky** [comment]
Before merge, let's make sure that the lecture notes are updated

**Review by karlhorky — APPROVED** [review]
LGTM, thanks!

**Pull request merged**